README.md

```diff
@@ -179,10 +179,167 @@ If you're calling a **GET** on `/entities` , you're asking for a list of entitie
+##### Reasoning
+
+Your endpoint should reflect what they do based on the HTTP Verb plus the endpoing name (noun). Therefore it makes sense that doing a **GET** to `/entities` retrieves a list of entities. On the contrary, doing a **GET** to `/get/entity` looks redundant and it's not self descriptive. 
+Let's try with another example: **POST** `/update_entity/:id`. That, semantically speaking, should be interpreted as creating/adding an entity to the `update_entity` set. If your intention was to update the entity with id `:id`, you should've used **PUT** or **PATCH** `/entities/:id` instead.
+
+---
+
+#### Plural nouns or singular nouns?
+
+> If your endpoint manages a list or a group of entities instead of just one then it should be named with a plural noun. If it always manages a single entity then it should be named with a singular noun.
+
+##### Examples
+
+###### Good
+
+> Getting comments from a post
+
+- **GET** `/posts/:id/comments`
+  The endpoint retrieves a list of comments for the post with the id you pass along in the url.
+
+> Getting the likes count of a post
+
+- **GET** `/posts/:id/like_count`
+  The endpoint retrieves the likes count of a post with the id you pass along in the url.
+
+> Deleting all the comments from a post
+
+- **DELETE** `/posts/:id/comments`
+  The endpoint deletes all the comments from the post with the id you pass along in the url.
+
+###### Bad
+
+> Getting a post
+
+- **GET** `/post/:id`
+  You're getting a post from a list/group of posts, so it should be plural `/posts/:id`.
+
+> Updating a post
+
+- **PUT** `/posts`
+  You're trying to update a post but you're not passing the id on the url. Passing it along in the request body is wrong.
+
+> Deleting a comment from a post
+
+- **DELETE** `/posts/:post_id/comment/:id`
+  Here you have a list/group of comments so it should be plural `/posts/:post_id/comments/:id`. Notice how I changed `comment` for `comments`.
+
+##### Reasoning
+
+Your endpoint name should be clear about the fact that it affects a single object or a group/list of objects. If you do a **POST** to `/posts` , you'll find the object you created by doing a **GET** to `/posts/:id`. 
+
+---
+
+#### Query String VS. Request Body. Where to send the object?
+
+> Query Strings are only valid on **GET** requests and every parameter they include is or should be used as a filter on the data returned. The only scenario where a parameter goes in the path is to identify a resource. Every other paremeter (in **POST**, **PUT**, **PATCH**, etc) should go in the request body. Any parameter that doesn't match any of the previous rules should go in headers (authentication, API versioning, etc).
```

Comment:
You should precede this text with something like: _There are 4 places where you can add parameters in your HTTP requests: querystring, path, body and headers._
That way the rest of the sentences in the paragraph will not look totally disconnected from each other.